tests/test_edgeql_insert.py

```diff
@@ -5822,23 +5822,34 @@ async def test_edgeql_insert_cardinality_assertion(self):
 
     @tb.needs_factoring_weakly
     async def test_edgeql_insert_volatile_01(self):
-        # Ideally we'll support these versions eventually
-        async with self.assertRaisesRegexTx(
-                edgedb.QueryError,
-                "cannot refer to volatile WITH bindings from DML"):
-            await self.con.execute('''
-                WITH name := <str>random(),
-                INSERT Person { name := name, tag := name };
-            ''')
+        await self.con.execute('''
+            WITH name := <str>random(),
+            INSERT Person { name := name, tag := name };
+        ''')
 
-        async with self.assertRaisesRegexTx(
-                edgedb.QueryError,
-                "cannot refer to volatile WITH bindings from DML"):
-            await self.con.execute('''
-                WITH name := <str>random(),
-                SELECT (INSERT Person { name := name, tag := name });
-            ''')
+        await self.assert_query_result(
+            r'''
+                SELECT all(Person.name = Person.tag)
+            ''',
+            [True]
+        )
+
+    @tb.needs_factoring_weakly
```

Comment:
I'd like to see some more tests, since this is a fiddly area.

Definitely one with a WITH binding inside of a FOR loop.
Maybe one with it inside a nested INSERT also.